src/app/common/UpgradeCheck.vue

```diff
@@ -26,67 +27,17 @@
         </div>
       </template>
     </KAlert>
-  </div>
+  </DataSource>
 </template>
 
 <script lang="ts" setup>
-import { KAlert, KButton } from '@kong/kongponents'
-import { ref } from 'vue'
-
-import { useEnv, useKumaApi, useI18n } from '@/utilities'
-
-const kumaApi = useKumaApi()
+import { useEnv, useI18n } from '@/utilities'
 const env = useEnv()
 const { t } = useI18n()
-
-const latestVersion = ref('')
-const showNotice = ref(false)
-
-checkVersion(env('KUMA_VERSION'))
-
-// mostly taken from semver-compare
-const compare = (a: string, b: string) => {
-  const pa = a.split('.')
-  const pb = b.split('.')
-  for (let i = 0; i < 3; i++) {
-    const na = Number(pa[i])
-    const nb = Number(pb[i])
-    if (na > nb) return 1
-    if (nb > na) return -1
-  }
-  return 0
-}
-
-async function checkVersion(currentVersion: string): Promise<void> {
-  if (!currentVersion.match('^[0-9]+.[0-9]+.[0-9]+$')) {
-    return
-  }
-
-  try {
-    latestVersion.value = await kumaApi.getLatestVersion()
-  } catch (error) {
-    console.error(error)
-    return
-  }
-  if (latestVersion.value !== '') {
-    // compare the latest version to the currently running version
-    // but only if we were able to set the latest version in the first place.
-    const comparison = compare(latestVersion.value, currentVersion)
-    showNotice.value = comparison === 1
-  } else {
-    const timespan = 3 // months
-    const today = new Date()
-    const refDate = new Date('2020-06-03 12:00:00')
-    const later = new Date(refDate.getFullYear(), refDate.getMonth() + timespan, refDate.getDate())
-
-    // compare dates and handle the notice accordingly
-    showNotice.value = today.getTime() >= later.getTime()
```

Comment:
I didn't bother replicating this stuff. I chased this back a long time ago after changing some code here a while back and it seems weird and I think it was original meant to store something in localstorage.

Essentially it shows the upgrade message if we can't retrieve the remote version number, which I do in this PR now but without all the date checking stuff.